### PR TITLE
only check debug.enabled when automatically setting quiet mode

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/datproject/dat/issues/830

### CODE REVIEW

**Comment by ralphtheninja** [comment]
Could also change this back to the previous logic, but just keep the debug message that silent mode is activated (or make it a console.log)

**Comment by joehand** [comment]
Have you tried running this as implemented here? Curious what the output looks like.

**Comment by ralphtheninja** [comment]
datclone1
datclone2

**Comment by ralphtheninja** [comment]
And with `DEBUG` set to something unrelated

datclone3

**Comment by joehand** [comment]
Ya, I was worried about when it's set to something unrelated. `neat-log` kind of clobbers some debug output in weird ways (see random progress bar in places below).

<details>
❯ DEBUG=discovery* dat clone datproject.org/jhand/cat-mapper cats
Connecting to dat network...
  discovery-channel chan=abd696..0d join() +0ms
  discovery-channel whoami() succeeded, I am: { port: 3282, host: '184.100.247.232' } +164ms
  discovery-channel chan=abd696..0d dns announce +4ms
  discovery-channel chan=abd696..0d dht announce +1ms
  discovery-channel chan=abd696..0d dns discovery peer=172.19.0.3:41765 +87ms
  discovery-swarm connecting 172.19.0.3:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d retries=0 +0ms
  discovery-channel chan=abd696..0d dns discovery peer=104.197.167.83:46483 +4ms
  discovery-swarm connecting 104.197.167.83:46483@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d retries=0 +3ms
  discovery-channel chan=abd696..0d dns discovery peer=172.19.0.6:57559 +1ms
  discovery-swarm connecting 172.19.0.6:57559@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d retries=0 +1ms
  discovery-swarm onclose utp+tcp 104.197.167.83:46483@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=0 +58ms
  discovery-swarm onconnect 104.197.167.83:46483@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d type=tcp +1ms
  discovery-channel chan=abd696..0d dns discovery peer=104.197.167.83:46483 +89ms
  discovery-channel chan=abd696..0d dns discovery peer=45.55.78.106:41765 +1ms
  discovery-swarm connecting 45.55.78.106:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d retries=0 +30ms
  discovery-channel chan=abd696..0d dns discovery peer=104.197.61.181:3282 +0ms
  discovery-swarm connecting 104.197.61.181:3282@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d retries=0 +0ms
  discovery-channel chan=abd696..0d dns discovery peer=45.55.78.106:57559 +0ms
  discovery-swarm connecting 45.55.78.106:57559@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d retries=0 +0ms
  discovery-swarm onclose utp+tcp 104.197.61.181:3282@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=0 +58ms
  discovery-swarm onconnect 104.197.61.181:3282@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d type=tcp +0ms
dat v13.8.1
Created new dat in /Users/joe/tmp/cats/.dat
Cloning: 14 files (40 MB)

2 connections | Download 338 KB/s Upload 0 B/s

Downloading updates...
[------------------------------------------] 1.62%


Ctrl+C to Exit
  discovery-channel chan=1.1 MB/s Upload 0 B/s
=149.172.254.59:45136 +1s
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:57559 via=8===---------------------------------------] 7.78%
  discovery-channel chan=abd696..0d dht discovery peer=104.197.167.83:46483 via=89.25.44.134:21114 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.61.181:3282 via  discovery-channel chan=abd696..0d dht discovery peer=104.197.167.83:46483 via=213.89.234.163:57741 +1s
  discovery-channel chan=abd696..0d dht discovery peer=104.197.61.181:3282 via=31.130.77.174:49001 +13ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.167.83:46483 via=31.130.77.174:49001 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:57559 via=31.130.77.174:49001 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:41765 via=177.99.22.77:6885 +5ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:57559 via=177.99.22.77:6885 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.61.181:3282 via=177.99.22.77:6885 +1ms
  discovery-channel chan=abd696..0d dht disco9.08%
a=177.99.22.77:6885 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:41765 via=61.165.135.80:26889 +78ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:57559 via=65.50.17.8:53906 +91ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.167.83:46483 via=65.50.17.8:53906 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:41765 via=65.50.17.8:53906 +1ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:41765 via=81.109.14.249:6881 +6ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:57559 via=81.109.14.249:6881 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.61.181:3282 via=81.109.14.249:6881 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.167.83:46483 via=81.109.14.249:6881 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.167.83:46483 via=179.198.190.16:10020 +125ms
  di=--------------------------------------] 10.05%
=179.198.190.16:10020 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:57559 via=179.198.190.16:10020 +1ms
  discovery-swarm timeout 172.19.0.3:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d +3s
  discovery-swarm timeout 172.19.0.6:57559@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d +1ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:57559 via=177.97.88.199:62567 +292ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.167.83:46483 via=177.97.88.199:62567 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=104.197.61.181:3282 via=177.97.88.199:62567 +0ms
  discovery-channel chan=abd696..0d dht discovery peer=45.55.78.106:41765 via=177.97.88.199:62567 +0ms
  discovery-swarm timeout 45.55.78.106:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d +91ms
  discovery-swarm timeout 45.55.78.106:57559@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d +0ms
  discovery-swarm onclose utp+tcp 45.55.78.106:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=1 +153ms
  discovery-swarm onclose utp+tcp 172.19.0.3:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=1 +0ms
  discovery-swarm onclose utp+tcp 172.19.0.6:57586%
d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=1 +1ms
  discovery-swarm onclose u0 MB/s Upload 0 B/s
f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=1 +0ms
  dis=-------------------------------------] 12.16%
189.82.17.8:59551 +492ms                      1.51%
  discovery-channel chan=abd696..0d dht discovery peer=104.197.61.181:3282 via=189.82.17.8:59551 +1ms
  discovery-swarm connect695 KB/s Upload 0 B/s
5033a0409ee97a741ad437c3d2ce55280d retries=1 +1s
  discovery-swarm connecting 172.19.0.3:41765@abd696bacfad22f35e7e21976f0d6b5033a040==-----------------------------------] 17.18%
  discovery-swarm connecting 172.19.0.6:57559@ab97%
33a0409ee97a741ad437c3d2ce55280d retries=1 +1ms
  discovery-swarm connecti68 KB/s Upload 0 B/s
  discovery-swarm timeout 45.55.78.106:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d +3s
  discovery-swarm timeout 172.19.0.3:41765@abd6967%
0409ee97a741ad437c3d2ce55280d +2ms
  discovery-swarm timeout 172.19.0.6:57559@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d +0ms
  discovery-swarm onclose 32 KB/s Upload 0 B/s
f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=1 +507ms
  discovery-swarm onclose utp+tcp 172.19.0.3:41765@abd696bacfad22f35e7e21976f0d6b5033a=----------------------------------] 19.77%
  discovery-swarm onclose utp+tcp 45.55.78.106:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=1 +1ms
  discovery-swarm onclose 34 KB/s Upload 0 B/s
  discovery-swarm connecting 45.55.78.106:57559@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d retries=2 +999ms
  discove==--------------------------------] 25.12%
33a0409ee97a741ad437c3d2ce55280d retries=2 +1ms
  discovery-swarm connecting 45.55.78.106:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d retries=2 +1ms
  discovery-swarm timeout 60 KB/s Upload 0 B/s
3a0409ee97a741ad437c3d2ce55280d +3stries=2 +0ms
  discovery-swarm timeout 172.19.0.3:41765@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a74=-------------------------------] 26.26%
  discovery-swarm timeout 45.55.78.106:41765@abd61%
3a0409ee97a741ad437c3d2ce55280d +1ms
  discovery-swarm timeout521 KB/s Upload 0 B/s
  discovery-swarm onclose utp+tcp 172.19.0.6:57559@abd696bacfad22f35e7e21976f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=1 +503ms
  discovery-==-----------------------------] 30.96%
f0d6b5033a0409ee97a741ad437c3d2ce55280d will-requeue=1 +0ms
  discovery-swarm onclose utp+tcp 172.19.0.3:41765@abd696bacfad22f35e7e21976f0Exiting the Dat program...
^Cdiscovery-swarm onclose utp+tcp 45.55.78.106:57559@abd696bacfad22f35e7e21976
</details>

If debug is enabled then neat-log should just print out the whole output to console.log instead of using ansi-diff-stream. I've been meaning to update that and have a few other neat-log things to fix, so let me see if that works.

**Comment by joehand** [comment]
Cool! So we can solve this by combining your PR with how it was previously. neat-log works better than expected with debug. So here is what I'll do:

* quiet: set when `debug.enabled` - this will ensure only dat's debug output is printing
* debug: set when `!!process.env.DEBUG` - this will print dat regular output in between debug statements every `opts.logspeed` without clobbering other messages.

We need to pass `opts.debug` through to all the neat-log instances, I can go through and update those.

**Comment by joehand** [comment]
Updated, let me know what you think. 

It still uses `opts.quiet` for dat's debugging. This isn't ideal (since it is non-intuitive) but its nice because it'll be easier to read debugged output and we don't accidentally leak the keys when someone copy/pastes debug output.

File: bin/cli.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ var invalidNode = nodeMajorVer < NODE_VERSION_SUPPORTED
 if (invalidNode) exitInvalidNode()
 
 if (debug.enabled) {
-  debug('DEBUG enabled, enabling quiet mode')
+  debug('Dat DEBUG mode engaged, enabling quiet mode')
```
Comment (ralphtheninja) [inline bin/cli.js]: :+1:

**Comment by ralphtheninja** [comment]
@joehand Looks nice!

**Comment by ralphtheninja** [comment]
w00t

**Pull request merged**